src/kbmod/region_search.py

```diff
@@ -1,424 +1,659 @@
+        """
+        if ic_to_export is None:
+            # Export the current ImageCollection
+            ic_to_export = self.ic
 
-        Returns
-        -------
-        uris : `list[str]`
-            The list of URIs for the given data Ids.
-        """
-        if dataset_types is None:
-            if self.dataset_types is None:
-                raise ValueError("No dataset types specified")
-            dataset_types = self.dataset_types
-        if collections is None:
-            if self.collections is None:
-                raise ValueError("No collections specified")
-            collections = self.collections
-
-        if not self.is_parallel():
-            return self._get_uris_serial(data_ids, dataset_types, collections)
-
-        # Divide the data_ids into chunks to be processed in parallel
-        data_id_chunks = list(_chunked_data_ids(data_ids))
-
-        # Use a ProcessPoolExecutor to fetch URIs in parallel
-        uris = []
-        with ProcessPoolExecutor(max_workers=self.max_workers) as executor:
-            futures = [
-                executor.submit(
-                    self._get_uris_serial,
-                    chunk,
-                    dataset_types=dataset_types,
-                    collections=collections,
-                    butler=self.new_butler(),
-                    trim_uri_func=trim_uri_func,
-                )
-                for chunk in data_id_chunks
-            ]
-            for future in as_completed(futures):
-                uris.extend(future.result())
+        if len(ic_to_export) < 1:
+            raise ValueError(f"ImageCollection is empty, cannot export {ic_to_export}")
 
-        return uris
+        new_ic = ic_to_export.copy()
 
-    def get_center_ra_dec(self, region):
+        # Add the metadata about the guess distance used when choosing this ImageCollection
+        if guess_dist is not None:
+            new_ic.data["helio_guess_dist"] = guess_dist
+
+        # Add the metadata about the patch this ImageCollection represents.
+        if patch is not None:
+            if not isinstance(patch, Patch):
+                if not isinstance(patch, int):
+                    raise ValueError("Patch must be an integer or a Patch object")
+                patch = self.get_patches()[patch]
+
+            # Populate the patch information to construct the patch WCS
+            patch_wcs = patch.to_wcs()
+            new_ic.data["global_wcs"] = patch_wcs.to_header_string()
+            new_ic.data["global_wcs_pixel_shape_0"] = patch_wcs.pixel_shape[0]
+            new_ic.data["global_wcs_pixel_shape_1"] = patch_wcs.pixel_shape[1]
+
+        # Reset standardizer-related metadata in our ImageCollection.
+        new_ic.meta["n_stds"] = len(new_ic)
+        new_ic.data["std_idx"] = range(len(new_ic))
+
+        return new_ic
+
+    def get_image_collection_from_patch(self, patch, guess_dist=0.0, min_overlap=0, max_images=None):
```

Comment:
Just for readability, perhaps this method should go before `export_image_collection`? I'm assuming that the user would call this before they would call `export_image_collection`.